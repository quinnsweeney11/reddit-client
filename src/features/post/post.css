.post{
    border: 1px solid grey;
    margin-bottom: 15px;
    display:flex;
    justify-content: space-around;
    max-height:400px;
    overflow:scroll;
    background-color:#211d1d;
}

.info {
    margin: 10px 0 10px 10px;
}

.info-item{
    display:inline-block
}

.contents{
    flex-grow:4;
    margin:0 20px;
}

.selftext{
    padding-bottom:20px;
}

a {
    color: #54beff
}

.points {
    color: #FF5700;
}

.children {
    color:#519fcf;
}

.image{
    width:60%;
    padding-bottom:20px;
}

.post_image{
    max-height:none;
    min-height:400px;
}

.image-carousel {
    width: 60%;
}

.carousel{
    width:100%;
}