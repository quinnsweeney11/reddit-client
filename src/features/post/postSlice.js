import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

//Define the inital state of Post:
const initialState = {
    title: '',
    text: '',
    media: [],
    timestamp: '',
    points: '',
    children: ''
}

export const getLocalData = createAsyncThunk('post/fetchLocalData', async () => {
    const response = await fetch('fakedata.json').catch(err => {console.log(err)});
    return response
})


export const postSlice = createSlice({
    name:'post',
    initialState,
    reducers: {
        
    }
})