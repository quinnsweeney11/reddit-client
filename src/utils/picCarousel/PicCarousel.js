import React, { useState } from 'react';
import './picCarousel.css';


export function PicCarousel(props) {
    const [currentPic, setCurrentPic] = useState(0);
    function nextPic(){
        setCurrentPic(currentPic + 1);
    }

    function previousPic(){
        setCurrentPic(currentPic - 1);
    }

    return(
        <div className="PicCarousel">
            <div className="controls">
                {currentPic === 0 ? <button disabled>ðŸš«</button>: <button onClick={() => previousPic()}>ðŸ‘ˆ</button>}
                {currentPic === props.pics.length - 1 ? <button disabled>ðŸš«</button> : <button onClick={() => nextPic()}>ðŸ‘‰</button>}
            </div>
            <img className="shown-img" src={`https://i.redd.it/${props.pics[currentPic].media_id}.jpg`} alt="Carousel" />
        </div>
    )
}