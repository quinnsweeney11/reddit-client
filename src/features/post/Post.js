import React from "react";
import { useDispatch } from "react-redux";
import "./post.css";
import ArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import ChatBubbleIcon from "@material-ui/icons/ChatBubble";
import { PicCarousel } from "../../utils/picCarousel/PicCarousel";
import "../../../node_modules/video-react/dist/video-react.css";
import ReactHlsPlayer from "react-hls-player";

import { setGallery } from "./postSlice";

//Function that we use to format the text posts.
function htmlDecode(input) {
    var doc = new DOMParser().parseFromString(input, "text/html");

    return doc.documentElement.textContent;
}

export function Post(props) {
    const dispatch = useDispatch();
    if (props.data.secure_media && props.data.secure_media.reddit_video) {
        return (
            <div className="post post_video">
                <div className="info">
                    <div className="info-item points">
                        <ArrowUpwardIcon />
                        <p>{props.data.score}</p>
                    </div>
                    <br />
                    <div className="info-item children">
                        <ChatBubbleIcon />
                        <p>{props.data.num_comments}</p>
                    </div>
                </div>
                <div className="contents">
                    <div className="info-item metadata">
                        <p><span className="rslash">r/</span>{props.data.subreddit}</p>
                        <p><span className="uslash">u/</span>{props.data.author}</p>
                    </div>
                    <h2 className="title-text">{props.data.title}</h2>
                    <ReactHlsPlayer src={props.data.secure_media.reddit_video.hls_url} autoplay={false} controls={true} className="video-container"/>
                </div>
            </div>
        );
    } else if(props.data.secure_media && props.data.secure_media.oembed) {
        return (
            <div className="post post_gif">
                <div className="info">
                    <div className="info-item points">
                        <ArrowUpwardIcon />
                        <p>{props.data.score}</p>
                    </div>
                    <br />
                    <div className="info-item children">
                        <ChatBubbleIcon />
                        <p>{props.data.num_comments}</p>
                    </div>
                </div>
                <div className="contents">
                    <div className="info-item metadata">
                        <p><span className="rslash">r/</span>{props.data.subreddit}</p>
                        <p><span className="uslash">u/</span>{props.data.author}</p>
                    </div>
                    <h2>{props.data.title}</h2>
                    <div className="embedded_media" dangerouslySetInnerHTML={{__html: htmlDecode(props.data.secure_media.oembed.html)}}>
                    </div>
                </div>
            </div>
        )
    }

    if (props.data.post_hint === "image") {
        return (
            <div className="post post_image">
                <div className="info">
                    <div className="info-item points">
                        <ArrowUpwardIcon />
                        <p>{props.data.score}</p>
                    </div>
                    <br />
                    <div className="info-item children">
                        <ChatBubbleIcon />
                        <p>{props.data.num_comments}</p>
                    </div>
                </div>
                <div className="contents">
                    <div className="info-item metadata">
                        <p><span className="rslash">r/</span>{props.data.subreddit}</p>
                        <p><span className="uslash">u/</span>{props.data.author}</p>
                    </div>
                    <h2>{props.data.title}</h2>
                    <img
                        className="image"
                        alt="Post Content"
                        src={props.data.url}
                    />
                </div>
            </div>
        );
    }

    if (props.data.gallery_data) {
        return (
            <div className="post post_carousel">
                <div className="info">
                    <div className="info-item points">
                        <ArrowUpwardIcon />
                        <p>{props.data.score}</p>
                    </div>
                    <br />
                    <div className="info-item children">
                        <ChatBubbleIcon />
                        <p>{props.data.num_comments}</p>
                    </div>
                </div>
                <div className="contents">
                    <div className="info-item metadata">
                        <p><span className="rslash">r/</span>{props.data.subreddit}</p>
                        <p><span className="uslash">u/</span>{props.data.author}</p>
                    </div>
                    <h2>{props.data.title}</h2>
                    <PicCarousel
                        className="carousel"
                        pics={props.data.gallery_data.items}
                    />
                </div>
            </div>
        );
    }

    if (props.data.selftext === "") {
        return (
            <div className="post">
                <div className="info">
                    <div className="info-item points">
                        <ArrowUpwardIcon />
                        <p>{props.data.score}</p>
                    </div>
                    <br />
                    <div className="info-item children">
                        <ChatBubbleIcon />
                        <p>{props.data.num_comments}</p>
                    </div>
                </div>
                <div className="contents">
                    <div className="info-item metadata">
                        <p><span className="rslash">r/</span>{props.data.subreddit}</p>
                        <p><span className="uslash">u/</span>{props.data.author}</p>
                    </div>
                    <h2>{props.data.title}</h2>
                </div>
            </div>
        );
    }

    return (
        <div className="post">
            <div className="info">
                <div className="info-item points">
                    <ArrowUpwardIcon />
                    <p>{props.data.score}</p>
                </div>
                <br />
                <div className="info-item children">
                    <ChatBubbleIcon />
                    <p>{props.data.num_comments}</p>
                </div>
            </div>
            <div className="contents">
                <div className="info-item metadata">
                    <p><span className="rslash">r/</span>{props.data.subreddit}</p>
                    <p><span className="uslash">u/</span>{props.data.author}</p>
                </div>
                <h4>{props.data.title}</h4>
                <div
                    className="selftext"
                    dangerouslySetInnerHTML={{
                        __html: htmlDecode(props.data.selftext_html),
                    }}
                ></div>
            </div>
        </div>
    );
}
