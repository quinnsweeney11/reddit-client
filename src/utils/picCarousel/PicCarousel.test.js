import React from "react";
import renderer from "react-test-renderer";
import { PicCarousel } from "./PicCarousel";
import { carouselPost } from "../../fakeData/post";

describe("snapshot", () => {
  it("photo post", () => {
    const component = renderer.create(
      <PicCarousel
        className="carousel"
        pics={carouselPost.data.gallery_data.items}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
