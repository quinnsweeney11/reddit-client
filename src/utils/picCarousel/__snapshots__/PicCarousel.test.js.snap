// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot photo post 1`] = `
<div
  className="PicCarousel"
>
  <div
    className="controls"
  >
    <button
      disabled={true}
    >
      ðŸš«
    </button>
    <button
      onClick={[Function]}
    >
      ðŸ‘‰
    </button>
  </div>
  <img
    alt="Carousel"
    className="shown-img"
    src="https://i.redd.it/6hqhwec7y3171.jpg"
  />
</div>
`;
